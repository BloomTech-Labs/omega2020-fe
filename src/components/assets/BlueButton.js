import React from 'react';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import { blue } from '@material-ui/core/colors';
import { withStyles } from '@material-ui/core/styles';

const BlueButton = (props) => {
  return (
    <ColorButton
      type={props.type}
      href={props.href}
      size='large'
      className='href'
      onClick={props.click}
    >
      <Typography variant='body2' className='title'>
        {props.title}
      </Typography>
    </ColorButton>
  );
};

const ColorButton = withStyles((theme) => ({
  root: {
    paddingLeft: 60,
    paddingRight: 60,
    paddingTop: 15,
    paddingBottom: 15,
    color: theme.palette.getContrastText(blue.A700),
    backgroundColor: blue.A700,
    '&:hover': {
      backgroundColor: blue[800],
    },
  },
}))(Button);

export default BlueButton;
