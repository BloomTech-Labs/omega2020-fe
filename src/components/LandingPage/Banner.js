import React from 'react';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import Box from '@material-ui/core/Box';
import Skeleton from '@material-ui/lab/Skeleton';
import CardHeader from '@material-ui/core/CardHeader';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

import BlueButton from '../assets/BlueButton';
import games from '../../images/games.png';

const Banner = (props) => {
  const classes = useStyles();
  const { loading = false } = props;

  let viewPort = false;
  const md = useMediaQuery('(min-width: 1050px)');
  if (viewPort === md) {
    viewPort = true;
  } else {
    viewPort = false;
  }

  return (
    <Box className={classes.root}>
      <Box className={classes.mainContainer}>
        <CardHeader
          avatar={
            loading ? (
              <Skeleton
                className='bannerIMG'
                variant='rect'
                width={600}
                height={600}
              />
            ) : (
              <Box className={classes.headerBox}>
                {(() => {
                  switch (viewPort) {
                    case true:
                      return (
                        <Box style={{ textAlign: 'center' }}>
                          <Typography
                            className={classes.header}
                            style={{ fontSize: '1.3rem' }}
                            id='improve1'
                          >
                            Improve Your Skills
                            <br />
                            By Understanding Game Strategies
                          </Typography>
                          <br />
                          <Typography
                            className={classes.subheader}
                            id='improve2'
                          >
                            Learn to recognize key patterns and methods
                            <br />
                            to improve your speed and skill level
                          </Typography>
                          <br />
                          <br />
                          <br />
                          <br />
                        </Box>
                      );
                    default:
                      return (
                        <Box style={{ textAlign: 'center' }}>
                          <Typography
                            variant='h4'
                            className={classes.header}
                            id='bang'
                          >
                            Improve Your Skills By Understanding Game Strategies
                          </Typography>
                          <br />
                          <Typography
                            variant='h6'
                            className={classes.subheader}
                          >
                            Learn to recognize key patterns and methods to
                            improve your speed and skill level
                          </Typography>
                          <br />
                          <br />
                          <br />
                          <br />
                        </Box>
                      );
                  }
                })()}
              </Box>
            )
          }
        />
        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box>
                <BlueButton title={'Get Started Today'} href={'/tutorial'} />
              </Box>
            )
          }
        />

        <CardHeader
          action={
            loading ? (
              <Skeleton className={classes.text} width={60} height={55} />
            ) : (
              <Box className={classes.gameBoard}>
                {(() => {
                  switch (viewPort) {
                    case true:
                      return (
                        <Box className={classes.img}>
                          {/* <img src={games} alt='games' width='300' /> */}
                        </Box>
                      );
                    default:
                      return (
                        <Box className={classes.img}>
                          {/* <img src={games} alt='games' /> */}
                        </Box>
                      );
                  }
                })()}
              </Box>
            )
          }
        />
      </Box>
    </Box>
  );
};

const useStyles = makeStyles((theme) => ({
  root: {
    width: '100%',
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 100,
  },
  mainContainer: {
    display: 'flex',
    flexFlow: 'column wrap',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'left',
  },
  subheader: {
    // opacity: 0.3,
    color: '#A2A2A2',
  },
  img: {
    width: '90vw',
    height: '50vh',
    backgroundImage: `url(${games})`,
    backgroundRepeat: 'no-repeat',
    // backgroundSize: 'contain, cover',
    backgroundSize: 'cover',
    backgroundPosition: 'center',
  },
  gameBoard: {
    marginTop: 200,
  },
  space: {
    '& > *': {
      margin: theme.spacing(1),
    },
  },
}));

export default Banner;
