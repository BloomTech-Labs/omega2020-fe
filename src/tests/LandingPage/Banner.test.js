import React from 'react';
import renderer from 'react-test-renderer';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import '@testing-library/jest-dom/extend-expect';
import { Typography } from '@material-ui/core';
import BlueButton from '../../components/assets/BlueButton';
import Banner from '../../components/LandingPage/Banner';

configure({ adapter: new Adapter() });

describe('the information on the banner', () => {
  const rBanner = renderer.create(<Banner />); //r represent renderer
  const iBanner = rBanner.root; //i represent instance
  
  const wrapper = mount(<Banner />);
  const iWrapper = wrapper.instance();
  
  it('snapshot the info on th banner', () => {
    const banner = renderer.create(<Banner />).toJSON();
    expect(banner).toMatchSnapshot();
  });

  it('checks the prop in the BlueButton', () => {
    expect(iBanner.findByType(BlueButton).props.title).toBe(
      'Get Started Today'
    );
  });

  it('checks the root for info', () => {
    expect(wrapper.containsMatchingElement(
      <Typography> 
        Improve Your Skills 
        <br /> 
        By Understanding Game Strategies 
      </Typography>
    )).toBeTruthy();
    
    expect(wrapper.containsMatchingElement(
      <Typography>
        Learn to recognize key patterns and methods
        <br />
        to improve your speed and skill level
      </Typography>
    )).toBeTruthy();


    expect(iWrapper).toEqual(null)
    expect(iBanner.findByProps({id: 'improve1'}).props.children).toBeTruthy();
    expect(iBanner.findByProps({id: 'improve2'}).props.children).toBeTruthy();    

  });

});
