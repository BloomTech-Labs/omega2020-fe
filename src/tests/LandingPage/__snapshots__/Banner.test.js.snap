// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the information on the banner snapshot the info on th banner 1`] = `
<div
  className="MuiBox-root MuiBox-root-22 makeStyles-root-1"
>
  <div
    className="MuiBox-root MuiBox-root-23 makeStyles-mainContainer-2"
  >
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-avatar"
      >
        <div
          className="MuiBox-root MuiBox-root-24"
        >
          <div
            className="MuiBox-root MuiBox-root-25"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
              id="improve1"
              style={
                Object {
                  "fontSize": "1.3rem",
                }
              }
            >
              Improve Your Skills
              <br />
              By Understanding Game Strategies
            </p>
            <br />
            <p
              className="MuiTypography-root makeStyles-subheader-3 MuiTypography-body1"
              id="improve2"
            >
              Learn to recognize key patterns and methods
              <br />
              to improve your speed and skill level
            </p>
            <br />
            <br />
            <br />
            <br />
          </div>
        </div>
      </div>
      <div
        className="MuiCardHeader-content"
      />
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
      <div
        className="MuiCardHeader-action"
      >
        <div
          className="MuiBox-root MuiBox-root-26"
        >
          <a
            aria-disabled={false}
            className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-12 MuiButton-text href MuiButton-textSizeLarge MuiButton-sizeLarge"
            href="/tutorial"
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
          >
            <span
              className="MuiButton-label"
            >
              <p
                className="MuiTypography-root title MuiTypography-body2"
              >
                Get Started Today
              </p>
            </span>
            <span
              className="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="MuiCardHeader-root"
    >
      <div
        className="MuiCardHeader-content"
      />
      <div
        className="MuiCardHeader-action"
      >
        <div
          className="MuiBox-root MuiBox-root-27 makeStyles-gameBoard-5"
        >
          <div
            className="MuiBox-root MuiBox-root-28 makeStyles-img-4"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
